main.py

```diff
@@ -0,0 +1,243 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START gae_python37_app]
+#from flask import Flask
+import flask
+from google.cloud import datastore
+client = datastore.Client()
+import json
+
+# If `entrypoint` is not defined in app.yaml, App Engine will look for an app
+# called `app` in `main.py`.
+app = flask.Flask(__name__)
+
+def getID():
+    q = client.query(kind="Tournament")
+    results = list(q.fetch())
+    currentID = 0
+    for t in results:
```

Comment:
This is very slow to get ID,  can you come up faster way?  may be create a record to just store the latest id?